### PR TITLE
Feature/triangle

### PR DESCRIPTION
**Pull request by rolfvdhulst**
Adds a class for triangles with some of the most useful functions.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -45,7 +45,7 @@ add_library(roboteam_utils STATIC
         include/roboteam_utils/Statistics.h
         include/roboteam_utils/pid.h
         src/pid.cpp
-        )
+        include/roboteam_utils/Triangle.cpp include/roboteam_utils/Triangle.h)
```
Comment (mrlukasbos) [inline CMakeLists.txt]: formatting

File: include/roboteam_utils/Triangle.cpp [hunk 2]
```diff
@@ -0,0 +1,65 @@
+//
```
Comment (mrlukasbos) [inline include/roboteam_utils/Triangle.cpp]: The class misses documentation

**Review by mrlukasbos — DISMISSED** [review]
I miss the tests

**Comment by rolfvdhulst** [comment]
> 
> 
> I miss the tests

It's still a WIP ;)

File: src/Triangle.cpp [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    return ((((corner3.x - corner2.x)*(point.y - corner2.y) - (corner3.y - corner2.y)*(point.x - corner2.x)) >= 0)
+            == s_ab);
+}
+double Triangle::area() const {
+    return abs(
+            (corner1.x*(corner2.y - corner3.y) + corner2.x*(corner3.y - corner1.y) + corner3.x*(corner1.y - corner2.y))
+                    *0.5);
+}
+std::vector<LineSegment> Triangle::lines() const {
+    return std::vector<LineSegment>({LineSegment(corner1, corner2),
```
Comment (Yuhanun) [inline src/Triangle.cpp]:
```cpp
return { {corner1, corner2}, ..., ... };
```

**Review by Yuhanun — APPROVED** [review]

**Review by RobotJente — APPROVED** [review]
Looks good as well! Tests are good too :)

File: src/Triangle.cpp (hunk 3 above)
Comment (rolfvdhulst) [reply src/Triangle.cpp]: What would the effective difference be? I don't understand the reasoning behind this.

**Pull request merged**